### DISCUSSION TITLE
Can't find the way to upload a local image to imgur with right click

### DISCUSSION THREAD

**Discussion by Suzuki-Re** [gavvvr/obsidian-imgur-plugin#96] [Getting help]
Hello,

I still don’t understand how to upload local images to imgur, with right click and comman palette. Right clicking an image just zoom in (no related menu when right clicking link either) and the upload feature is not exposed in command palette.

Currently installed version is 2.6.3.

Thanks

**Comment by gavvvr** [comment]
Hi @Suzuki-Re 
You should put the cursor somewhere on the clickable **local** image element (technically between `![[` and `]]`).
Then you will get the following command palette option available:
<img width="481" alt="image" src="https://github.com/user-attachments/assets/a88baf61-5857-48a0-b4f8-bc6dadbaffa7">

There is also a right-click option on the bottom of context menu:
<img width="309" alt="image" src="https://github.com/user-attachments/assets/c65d3a38-51cb-4881-ba87-a1c93e62ebee">

**Comment by Suzuki-Re** [comment]
Thank you for a quick reply!

Yes, it shows the option when hovering over a image link in the root but not for a image file in folder.

An image file in the root
![Obsidian_uQu4imPxhl](https://github.com/user-attachments/assets/07ff8b72-35bf-434e-9e57-bc28ff1a0251)

An image file in the folder (path)
![Obsidian_nUgChH0mxx](https://github.com/user-attachments/assets/bef0e8db-7657-459a-a518-ced9ef9f9cf4)

Am I missing something obvious?

**Reply by gavvvr** [reply]
Hi @Suzuki-Re 

> but not for a image file in folder

At the moment there is no option to upload local image from vault tree right-click-context menu, only from editor context menu.

Upload from files tree can potentially be supported, but here is the design question for file tree context menu feature: If the local image you would want to upload is not used by any note,  where the remote URL of just uploaded local image should go? Would love to hear your ideas about that.
